궁금한것 동물을 사육사에 어떻게 넣었는지
사육사와 직원의 관계에 대해서
그

MAP은 key와 value가 있다
key는 원하는 값 value는 찾는값을 적으면 되나...?

원하는곳을 바로 찾아간다.


이제 관리 프로그램 만들면 된다!


try catch 는 프로그램 흐름(try)에서
예외가 나는 상황이 나면
예외를 처리 하는 (catch) 문장으로 넘어가는것 즉 정상흐름에서 예외를 만나면
 catch의 흐름으로 넘어가는 것이다.
이때 catch 파라미터()에 예외타입 을 입력하며 변수이름을 지정한다
예외타입의 객체가 형성 된 것이며 그 객체의 메소드 또한 존재한다.
어디서 오류가 난지 알 수 있는 메소드도 있고 메세지를 등록해서 보내는 메소드 들도 있다.


catch Exception은 상속이다.

NullPointerException 또한 부모로 Exception을 가진다

Exception은 모든 예외를 다 처리 한다.
만약 Exception을 하고 NullPointerException을 뒤에 넣으면
NullPointerException은 컴파일 오류가 난다.

예외가 실행 되던지 안되던지 무조건  실행되는 문장들은 finally 안에 있다.  
예외를 못잡고 오류가 난다 해도 finally는 실행이 된다.

잘못했을떄 사유서를 쓰는데
사유서를 보는것이 catch로 보는것이다.

예외처리를 하지 않으면 컴파일 오류가 난다.

Exception 안에는 RunTimeException이 있다.

실행중에 발생할 수 있는 예외들을 try catch를 요구하지 않고 실행 가능하게끔 RuntimeException 객체를 만든다.




예외일때만 왜 안된지를 아는 (return true 했던것 처럼!)

boolean은 if일때는 true 반환
else 일때 false 반환이라 한다면

출금할때만 예외가 


st.hasMoreElements();
// boolean값을 반환한다. 
나누는 값이 있냐를 확인해서 true와false를 반환해준다. 
만약 토큰이 3개뿐인데 4개를 출력하라하면 출력메소드는 작동이 된다. 
하지만 토큰은 3개뿐이라 예외상황이 발생한다






